internal/authz/authz.go

```diff
@@ -0,0 +1,237 @@
+	case ListStores:
+		return CanCallListStores, nil
+	case CreateStore:
+		return CanCallCreateStore, nil
+	case GetStore:
+		return CanCallGetStore, nil
+	case DeleteStore:
+		return CanCallDeleteStore, nil
+	case Expand:
+		return CanCallExpand, nil
+	case ReadChanges:
+		return CanCallReadChanges, nil
+	default:
+		return "", fmt.Errorf("unknown api method: %s", apiMethod)
+	}
+}
+
+// Authorize checks if the user has access to the resource.
+func (a *Authorizer) Authorize(ctx context.Context, clientID, storeID, apiMethod string) error {
+	relation, err := a.getRelation(apiMethod)
+	if err != nil {
+		return err
+	}
+
+	authorized, err := a.individualAuthorize(ctx, clientID, relation, a.getStore(storeID), &openfgav1.ContextualTupleKeys{})
+	if err != nil {
+		return err
+	}
+
+	if !authorized {
+		return status.Error(codes.Code(ErrorResponse.GetCode()), ErrorResponse.GetMessage())
+	}
+
+	return nil
+}
+
+// AuthorizeCreateStore checks if the user has access to create a store.
+func (a *Authorizer) AuthorizeCreateStore(ctx context.Context, clientID string) error {
+	relation, err := a.getRelation(CreateStore)
+	if err != nil {
+		return err
+	}
+	authorized, err := a.individualAuthorize(ctx, clientID, relation, a.getSystem(), &openfgav1.ContextualTupleKeys{})
+	if err != nil {
+		return err
+	}
+
+	if !authorized {
+		return status.Error(codes.Code(ErrorResponse.GetCode()), ErrorResponse.GetMessage())
+	}
+
+	return nil
+}
+
+// AuthorizeListStores checks if the user has access to list stores.
+func (a *Authorizer) AuthorizeListStores(ctx context.Context, clientID string) error {
+	relation, err := a.getRelation(ListStores)
+	if err != nil {
+		return err
+	}
+
+	authorized, err := a.individualAuthorize(ctx, clientID, relation, a.getSystem(), &openfgav1.ContextualTupleKeys{})
+	if err != nil {
+		return err
+	}
+	if !authorized {
+		return status.Error(codes.Code(ErrorResponse.GetCode()), ErrorResponse.GetMessage())
+	}
+
+	return nil
+}
+
+// ListAuthorizedStores returns the list of stores that the user has access to.
+func (a *Authorizer) ListAuthorizedStores(ctx context.Context, clientID string) ([]string, error) {
+	err := a.AuthorizeListStores(ctx, clientID)
+	if err != nil {
+		return nil, err
+	}
+
+	req := &openfgav1.ListObjectsRequest{
+		StoreId:              a.config.StoreID,
+		AuthorizationModelId: a.config.ModelID,
+		User:                 a.getApplication(clientID),
+		Relation:             CanCallGetStore,
+		Type:                 StoreType,
+	}
+
+	// Disable authz check for the list objects request.
+	ctx = authcontext.ContextWithSkipAuthzCheck(ctx, true)
+	resp, err := a.server.ListObjects(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+	authcontext.ContextWithSkipAuthzCheck(ctx, false)
+
+	return resp.GetObjects(), nil
+}
+
+func (a *Authorizer) individualAuthorize(ctx context.Context, clientID, relation, object string, contextualTuples *openfgav1.ContextualTupleKeys) (bool, error) {
```

Comment:
Do we foresee needing to pass contextual tuples? if not, can we remove the last parameter as it's never being used at the moment?

Reply:
It's being used in https://github.com/openfga/openfga/pull/1914